toolchain/check/handle_name.cpp

```diff
@@ -204,6 +230,13 @@ auto HandleParseNode(Context& context,
   return true;
 }
 
+auto HandleParseNode(Context& context,
+                     Parse::KeywordNameQualifierWithoutParamsId /*node_id*/)
+    -> bool {
+  context.decl_name_stack().ApplyNameQualifier(PopNameComponent(context));
```

Comment:
This would mean we had `destroy.Something` right? I see that we end up rejecting it for the `Something`, but would the diagnostic be better if we rejected `destroy.` earlier, like maybe even in parse?

Reply:
I gave some detail on why we don't try to reject during parse in the parse PR:
https://github.com/carbon-language/carbon-lang/pull/5045#discussion_r1976186690